src/NLog/Common/AsyncHelpers.cs

```diff
@@ -379,5 +386,17 @@ internal static bool WaitForDispose(this Timer timer, TimeSpan timeout)
             }
             return true;
         }
+
+        /// <summary>
+        /// Forward declare of system delegate type for use by other classes
+        /// </summary>
+        internal struct AsyncTask
```

Comment:
Proposal, unnest and move to internal.

Reply:
Well it really belongs to AsyncHelpers.

Reply:
It's now inconsistent when it's a class in the internal namespace and when it's nested.

Reply:
Fixed